src/Stratis.Bitcoin/P2P/Peer/NetworkPeerServer.cs

```diff
@@ -164,23 +172,20 @@ public void Listen(int maxIncoming = 20)
             {
                 try
                 {
-                    int taskCount = Interlocked.Increment(ref this.unfinishedTasks);
-                    this.logger.LogTrace("Starting accepting task, there are {0} unfinished tasks now.", taskCount);
+                    this.serverCancel.Token.ThrowIfCancellationRequested();
```

Comment:
You don't need this if the condition is in the loop statement. 
` this.serverCancel.Token.ThrowIfCancellationRequested();`

Reply:
this method is improved in later PR, but you are correct again